### DISCUSSION TITLE
Unable to show orders in front-end

### DISCUSSION THREAD

**Discussion by de-raaf-media** [duncanmcclean/simple-commerce#1042] [Help]
In Simple Commerce 5 I show the orders in the front-end. But for some reason. In the front-end I can no longer see the orders.

$orders = $customer->orders();

I also noticed that the orders column remains empty when placing an order. In v6 I switched to database orders. They were in the database previously but as entries because I used the eloquent driver for this.

**Comment by duncanmcclean** [comment]
Where is this code from? Is this your own code?

```php
$orders = $customer->orders();
```

**Comment by de-raaf-media** [comment]
Well. It is my implementation of the PHP api: https://simple-commerce.duncanmcclean.com/php-api#php-api.

```
use DuncanMcClean\SimpleCommerce\Facades\Customer;
...
$customer = Customer::find(auth()->id());
...
$orders = $customer->orders();
```

**Comment by duncanmcclean** [comment]
What does the `content` array look like in your `simple-commerce.php` config?

**Comment by de-raaf-media** [comment]
Hi, it looks like this:

```
'content' => [
    'customers' => [
        'repository' => \DuncanMcClean\SimpleCommerce\Customers\UserCustomerRepository::class,
        'collection' => 'customers',
    ],

    'orders' => [
        'repository' => \DuncanMcClean\SimpleCommerce\Orders\EloquentOrderRepository::class,
        'model' => \DuncanMcClean\SimpleCommerce\Orders\OrderModel::class,
    ],

    'products' => [
        'repository' => \DuncanMcClean\SimpleCommerce\Products\EntryProductRepository::class,
        'collection' => 'products',
    ],
],
```

**Reply by duncanmcclean** [reply]
Seems to be working fine on my sandbox site: 

![CleanShot 2024-04-08 at 20 59 32](https://github.com/duncanmcclean/simple-commerce/assets/19637309/7afb492a-4fec-4598-b315-7a71bf139444)

If you're using an off-site gateway (like Mollie or Stripe Payment Elements), is the webhook setup & working properly? If it's not, the order won't be marked as paid and the order won't be added to the customer's profile.

**Comment by de-raaf-media** [comment]
Well for now I use the dummy gateway. Unfortunately I am unable to checkout. After filling in the payment details and clicking submit the next error is presented:

str_replace(): Argument #2 ($replace) must be of type string when argument #1 ($search) is a string.

I manually changed the order to placed and paid (statuses seem to be correct in the CP).

Maybe this is all related to the order not being saved properly somehow. I did notice the orders are added to the database but the user orders column remains empty. In my V5 version the are orders in the users column. Maybe that's part of the problem (Edit: I now realize that you said that the order won't be added to the profile if payment is not executed properly).

**Reply by duncanmcclean** [reply]
Yes, the orders will only be added to the customer/user if the checkout process is completed succesfully. It won't be triggered by manually changing the order status (although, now i think about it, maybe it should but it doesn't right now 😓 )

Can you provide the full stack trace for that str_replace error so I can tell where it's coming from?

**Comment by de-raaf-media** [comment]
I will send you an e-mail with the full stack trace.

**Comment by de-raaf-media** [comment]
Problem solved. The orders are visible after checkout. The checkout wasn't working because of an error in my template. Thanks for helping me out with this Duncan!

**Reply by duncanmcclean** [reply]
Awesome, glad to hear it!
